Zotero/Scenes/Detail/PDF/ViewModels/PDFReaderActionHandler.swift

```diff
@@ -1629,13 +1629,21 @@ final class PDFReaderActionHandler: ViewModelActionHandler, BackgroundDbProcessi
                 throw PDFReaderState.Error.documentEmpty
             }
 
-            let isDark = viewModel.state.interfaceStyle == .dark
             let key = viewModel.state.key
+            let (item, liveAnnotations, storedPage) = try loadItemAnnotationsAndPage(for: key, libraryId: viewModel.state.library.identifier)
+
+            if let url = viewModel.state.document.fileURL, let md5 = md5(from: url), item.backendMd5 != md5 {
+                update(viewModel: viewModel) { state in
+                    state.changes = .md5
+                }
```

Comment:
This part is similar to `case .change` in nested function `observe` below. It can be refactored to a private method, or even better another nested function.